src/map.rs

```diff
@@ -216,19 +217,30 @@ impl Map {
 
         Self::draw_ui(my_id, game_state, &mut powerup_sbs, ctx, assets);
 
-        graphics::draw(ctx, &background_sb, (na::Point2::new(0., 0.),)).unwrap();
-        graphics::draw(ctx, &smoke_sb, (na::Point2::new(0., 0.),)).unwrap();
+        // the offset which causes the camera to shake upon getting hit
+        let mut hit_offset = na::Vector2::new(0., 0.);
+        if hit_effect_timer > 0. {
+            hit_offset = na::Vector2::new(
+                (hit_effect_timer*constants::HIT_SHAKE_SPEED*0.9).sin()
```

Comment:
0.9 and 1.1 are magic constants. Why is x and Y different?

Reply:
The reason why I didn't put these in constants is because I didn't want these to be tweaked. The reason why they are there is to make the shake a bit more irregular, if x=y then it will shake along the diagonal.